### ISSUE TITLE
Some way to prevent cursor flickering

### ISSUE THREAD

**Issue by monkoose** [vim/vim#11918]
TL;DR some plugins which do heavy computation + redrawing causing cursor to flicker

[screencast example](https://user-images.githubusercontent.com/6261276/215587343-9c2ccc2c-0424-43c9-b49f-f2798ada1acd.mp4)

I'm using a combination of pretty popular plugins: [ALE](https://github.com/dense-analysis/ale) + [CoC](https://github.com/neoclide/coc.nvim) which have some autocmds on CursorHold, InsertLeave, etc, which process some lsp/linting behind the scenes. They are working fine, but also a cause of the cursor flickering (starting visible in some big files with 1000+ lines) as shown in the screencast above.

This behavior can be replicated with something like this
```vim
vim9script

def HeavyComputation(_): number
  var sum = 0
  redraw
  for _ in range(200000)  # adjust loop size if flickering still not visible on your machine or you can set it very big to completely hide the cursor for the time of computation
    sum += 2
  endfor
  return sum
enddef

augroup FlickerReplicate
  autocmd!
  autocmd CursorHold * timer_start(1, HeavyComputation)
  autocmd CursorHold * timer_start(20, HeavyComputation)
  autocmd CursorHold * timer_start(40, HeavyComputation)
augroup END

defcompile
```

My assumption this is happening when `redraw` follows some heavy computation. If we remove `redraw` from this code, then the flickering is gone. Is there a reason for such behavior that the cursor is hidden until this computation happens? Can it be prevented somehow?
